pydantic_ai_slim/pydantic_ai/models/outlines.py

```diff
@@ -0,0 +1,500 @@
+            if key in ['use_tqdm', 'lora_request', 'priority']:
+                continue
+            setattr(sampling_params, key, model_settings.pop(key, None))
+
+        model_settings['sampling_params'] = sampling_params
+
+        return model_settings
+
+    def _format_prompt(self, messages: list[ModelMessage]) -> Chat:  # noqa: C901
+        """Turn the model messages into an Outlines Chat instance."""
+        chat = Chat()
+        for message in messages:
+            if isinstance(message, ModelRequest):
+                for part in message.parts:
+                    if isinstance(part, SystemPromptPart):
+                        chat.add_system_message(part.content)
+                    elif isinstance(part, UserPromptPart):
+                        if isinstance(part.content, str):
+                            chat.add_user_message(part.content)
+                        elif isinstance(part.content, Sequence):
+                            if not all(isinstance(item, str) for item in part.content):
+                                # The expected format of the assets is not compatible
+                                # between Pydantic AI and Outlines yet. Outlines will
+                                # widen support in the future.
+                                raise UserError('Outlines does not support multi-modal user prompts yet.')
+                            chat.add_user_message(str(part.content))
+                        else:
+                            assert_never(part.content)
+                    elif isinstance(part, RetryPromptPart):
+                        chat.add_user_message(str(part.content))
+                    elif isinstance(part, ToolReturnPart):
+                        raise UserError('Tool calls are not supported for Outlines models yet.')
+                    else:
+                        assert_never(part)
+            elif isinstance(message, ModelResponse):
+                for part in message.parts:
+                    if isinstance(part, TextPart):
+                        chat.add_assistant_message(part.content)
+                    elif isinstance(part, ThinkingPart):
+                        # NOTE: We don't send ThinkingPart to the providers yet.
+                        pass
+                    elif isinstance(part, ToolCallPart | BuiltinToolCallPart | BuiltinToolReturnPart):
+                        raise UserError('Tool calls are not supported for Outlines models yet.')
+                    else:
+                        assert_never(part)
+            else:
+                assert_never(message)
+        return chat
+
+    def _process_response(self, response: str) -> ModelResponse:
+        """Turn the Outlines text response into a Pydantic AI model response instance."""
+        return ModelResponse(
+            parts=cast(
+                list[ModelResponsePart], split_content_into_text_and_thinking(response, self.profile.thinking_tags)
```

Comment:
Is "Some local models run through Outlines include a reasoning part in their output. In that case, it will be handlded by Pydantic AI without the need to specifically enable it." in the docs correct? "reasoning part" sounds like it refers to a dedicated reasoning field, not just parsing think tags from text output